### ISSUE TITLE
The example in "Usage" hangs with google drive

### ISSUE THREAD

**Issue by emanuele** [DanielDent/git-annex-remote-rclone#6]
I've just tried to run the example explained in [Usage](https://github.com/DanielDent/git-annex-remote-rclone#usage), on a google drive remote. Unfortunately it hangs indefinitely after printing `initremote gdrive (encryption setup)`. Details:
- I'm using rclone 1.30, git-annex 6.20160706 and the last git-annex-remote-rclone. All of them in PATH.
- I've configured rclone to work with my google drive (`rclone config` , and created the gdrive_rclone remote) and tested it by rcloning some stuff (`rclone sync LOCALDIR gdrive_rclone:`): it works.
- I've created a small git-annex repository on my laptop, following the walkthrough.
- I've tried to create a special remote using your git-annex-remote-rclone, following the example in README/Usage, specifically:
  `# git annex initremote gdrive type=external externaltype=rclone target=gdrive_rclone prefix=git-annex chunk=50MiB encryption=shared mac=HMACSHA512 rclone_layout=lower`
  but I just get this message: `initremote gdrive (encryption setup)` and the process hangs forever.

Could you please help me to sort out the issue?

**Comment by DanielDent** [comment]
Great detail, it should make it easier to help.

During the initremote step, git-annex makes a call out to GPG along with making a call to the remote. Perhaps some aspect of your configuration is causing GPG to hang waiting on input from you?

Try running  `git annex --debug initremote gdrive type=external externaltype=rclone target=gdrive_rclone prefix=git-annex chunk=50MiB encryption=shared mac=HMACSHA512 rclone_layout=lower` (I added `--debug`) and pasting the output to this thread.

**Comment by DanielDent** [comment]
Also, I just noticed that 6.20160706 is not a released version of git-annex. Joey recently fixed a remote related bug. In addition to using the `--debug` flag, please also try git-annex version: 6.20160613-g35dbe, which is the latest version that has gone through a successful CI run (A CI process runs periodically against the latest stable releases of rclone and git-annex).

**Comment by emanuele** [comment]
Thank you for your prompt reply. Indeed the issue was in using GPG and, in the end, there was no bug: `gpg --gen-random 2` just took forever on the computer on which I ran `git-annex initremote...`, because the entropy pool was starving and the computer was unused (so no hope to collect more entropy in a decent amount of time). In the failed attempt, I waited more than 30 minutes and then interrupted the process. Now I'm filling the entropy pool by forcefuly interacting with such computer (mouse, keyboard, etc.) and, in the end, it worked.
It might be better to warn users of this aspect.

Notice that I can confirm that the latest (unreleased) git-annex, i.e. 6.20160706, works too.

Another issue (or potential one) popped-up: git-annex + rclone + git-annex-remote-rclone work only if the rclone configuration password is not set. In that case I was able to init the remote and to copy (and get) the content to (from) the git-annex repository on gdrive. Differently, with the rclone password set, each step failed - presumably because the interactive request of the password, from rclone, is not supported. I can open a new github issue on this one, to keep things distinct.

**Comment by DanielDent** [comment]
Glad I was able to help. Thanks for the update.

GPG is quite picky about it's entropy sources, documenting this seems like the right approach.

Regarding a password-protected rclone configuration: I haven't tried this configuration, and it is not supported at this time. For now I can document this limitation. A new github issue to add this feature would be great, though for what it's worth I likely won't have time to work on it myself any time soon.

I am curious: what lead you to use this remote for Google Drive? There is a Google Drive specific remote linked on the git-annex wiki. When I released this remote I only updated links on the git-annex wiki for providers which did not have an existing remote built specifically for them, but looking more closely at the Google Drive remote the project looks somewhat dead. Have you tried that remote / do you think Google Drive users should be directed to this remote?

**Comment by emanuele** [comment]
Thank you for the improvements in the documentation.
I am investigating your git-annex-remote-rclone because I consider rclone a pretty mature project and then Joey [mentions](http://git-annex.branchable.com/tips/googledriveannex/) that the google drive remote project currently proposed in git-annex (googledriveannex) seems not developed anymore, as you correctly report. I haven't tried it at the moment. Would you suggest  [googledriveannex](https://github.com/TobiasTheViking/googledriveannex) over yours for the moment?

**Comment by emanuele** [comment]
OK, I've just tried [googledriveannex](https://github.com/TobiasTheViking/googledriveannex) but it does not work for me (see last comment [here](http://git-annex.branchable.com/tips/googledriveannex/)). There seems to be an issue at the initialization phase (OAUTH) of the special remote. Differently, your git-annex-remote-rclone seems to work well. A this point I think Google Drive users should be directed to your external special remote.
